### PR TITLE
SYSCHK changes

### PR DESCRIPTION
**Pull request by b1ackmai1er**
Change SYSCHK message to all display the error code after the register dump.
Changed SYSCHK call to a macro to ensure error code value is always in a known location

### CODE REVIEW

File: Source/HBIOS/simrtc.asm [hunk 1]
```diff
@@ -55,9 +55,7 @@ SIMRTC_DISPATCH:
 	JP	Z,SIMRTC_SETALM	; SET ALARM 
 	DEC	A
 	JP	Z,SIMRTC_DEVICE	; REPORT RTC DEVICE INFO
-	CALL	SYSCHK
-	LD	A,ERR_NOFUNC
-	OR	A
+	SYSCHKERR(ERR_NOTFUNC)
```
Comment (wwarthen) [inline Source/HBIOS/simrtc.asm:58]: Please fix: ERR_NOTFUNC --> ERR_NOFUNC

Comment (wwarthen) [reply Source/HBIOS/simrtc.asm:58]: Thanks for all this work Phil.  I marked one small issue that causes a build error.  Can you fix this before I merge?

Thanks,

Wayne

File: Source/HBIOS/hbios.inc [hunk 2]
```diff
@@ -156,6 +156,111 @@ ERR_TIMEOUT		.EQU	-11	; DEVICE TIMEOUT
...
+DIAG_DISP	.EQU	DIAG_PROG	; DEFAULT
+;
+#IF (DIAG_DISP == DIAG_PROG)
+DIAG_00 .EQU 00000000B	
+DIAG_01 .EQU 00000001B 
+DIAG_02 .EQU 00000011B 
+DIAG_03 .EQU 00000111B 
+DIAG_04 .EQU 00001111B 
+DIAG_05 .EQU 00011111B 
+DIAG_06 .EQU 01111111B
```
Comment (wwarthen) [inline Source/HBIOS/hbios.inc:178]: It looks like you skipped an LED going from DIAG_05 TO DIAG_06.

File: Source/HBIOS/hbios.asm [hunk 3]
```diff
@@ -3499,7 +3504,7 @@ HB_DSKFN:
 HB_DSKFNADR	.EQU	$+1
 	CALL	PANIC			; READ ONE SECTOR
 #IF (DIAGENABLE & DIAGDISKIO)
-	DIAG(0)				; CLEAR DIAG LEDS
+	DIAG(DIAG00)				; CLEAR DIAG LEDS
```
Comment (wwarthen) [inline Source/HBIOS/hbios.asm:3507]: Looks like this should be DIAG_00.

**Comment by wwarthen** [comment]
Thanks Phil.  Sorry, but one of the new changes breaks the build.  Can you take a look?

Thanks!

Wayne

**Comment by b1ackmai1er** [comment]
hi wayne, fix submitted, thanks.

**Comment by wwarthen** [comment]
Thanks Phil.  Sorry for the delay, but this all looks great.

-Wayne

**Pull request merged**